### PR TITLE
Objective-C: Use an OAuth library via CocoaPods

### LINKED ISSUE #57: Objective-C OAuth library does not use ARC
@MatthewHendy writes in #32 :
>In my build settings the 'Objective-C Automatic Reference Counting' flag is set to yes, yet I have 94 errors relating to 'release' , 'autorelease' , ARC restrictions, and ARC casting rules. I copied this code into my project and double checked my ARC flag and it doesn't run. If I turn off ARC then it runs fine.

First step: Repro and find a few of these 94 errors.

@MatthewHendy: Do you have 2-3 example errors we could look at?

### PR DESCRIPTION
**Pull request by SSheldon**
The current Objective-C sample has an old version of OAuthConsumer copied in, and it trips a lot of people up because it isn't very friendly. One of the biggest issues people run into is that the library isn't updated to use ARC.

Additionally, copying external libraries into your code isn't the best way to integrate them; nowadays lots of Objective-C projects are using CocoaPods, so the example will feel more modern by using it. And it's really convenient!

After looking around at some of the availabe OAuth pods, [TDOAuth](https://github.com/tweetdeck/TDOAuth) seemed like one of the nicest and simplest choices.

The sample works the same after this change.

Fixes #57.

### CODE REVIEW

**Comment by tomelm** [comment]
"89 additions and 3,329 deletions" <3

File: v2/objective-c/.gitignore [hunk 1]
```diff
@@ -8,10 +12,17 @@ build/
...
-DerivedData
+*.xcuserstate
+*.xcscmblueprint
+
+# Obj-C/Swift specific
 *.hmap
 *.ipa
-*.xcuserstate
 
+# CocoaPods
```
Comment (mittonk) [inline v2/objective-c/.gitignore]: This looks great.

You've got a TODO around the readme --- if you add that, or decide you don't want it, I'd be happy to give a ship-it.

Comment (SSheldon) [inline v2/objective-c/.gitignore]: @mittonk, readme is updated now! If it looks good to you, it should be ready to go.

**Comment by mittonk** [comment]
:shipit:

**Pull request merged**